components/src/dynamo/sglang/afd_metrics.py

```
@@ -0,0 +1,503 @@
+                # Need to reduce ratio for lower latency
+                optimal_ratio = max(1, optimal_ratio - 1)
+        
+        return optimal_ratio
+    
+    def get_resource_utilization(self) -> Dict[str, Any]:
+        """Analyze resource utilization across workers.
+        
+        Returns:
+            Utilization analysis
+        """
+        metrics = self.collector.get_aggregate_metrics()
```

Ruff (0.15.2)

[error] Local variable `metrics` is assigned to but never used

Remove assignment to unused variable `metrics`

(F841)